### PR TITLE
GCC 8.1 recipe

### PR DESCRIPTION
**Pull request by extrowerk**
builds, packages fine

### CODE REVIEW

**Comment by extrowerk** [comment]
I had the first random "ld" crash in x86_elf_aligned_common.
The next run finished correctly. Somebody should check that part in the patchset:
https://github.com/haikuports/haikuports/pull/2542/files#diff-7251456114e2453293249aa3c50e69d1R3175

I'll post the backtrace later.

**Comment by extrowerk** [comment]
So the backtrace: https://gist.github.com/extrowerk/ec2a9e18a6b00cd06ea03f1bc2ce70db

File: sys-devel/gcc/gcc8-8.1.0_2018_05_07.recipe [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+	lib:libgfortran$secondaryArchSuffix = $libgfortranLibVersion compat >= $libgfortranSoVersion
+	lib:libgomp$secondaryArchSuffix = $libgompLibVersion compat >= $libgompSoVersion
+	lib:libquadmath$secondaryArchSuffix = $libquadmathLibVersion compat >= $libquadmathSoVersion
+	lib:libssp$secondaryArchSuffix = $libsspLibVersion compat >= $libsspSoVersion
+	lib:libstdc++$secondaryArchSuffix = $libstdcxxLibVersion compat >= $libstdcxxSoVersion
+	lib:libsupc++$secondaryArchSuffix = $portVersion compat >= 7
+	"
+REQUIRES_syslibs="
+	haiku$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
```
[the remark this answers is not in the document]
Comment (waddlesplash) [reply sys-devel/gcc/gcc8-8.1.0_2018_05_07.recipe]: Probably fine. We need it for bootstrapping anyway.

**Comment by waddlesplash** [comment]
I *think* that's the same LD crash I saw before disabling the older ld in https://github.com/haikuports/haikuports/commit/1e51e1c8bfd1a9e6a315adccf063d667c0eb7ad0

**Comment by waddlesplash** [comment]
Ah, no it isn't, that was #1750. @extrowerk, try updating binutils to the latest version also along with the gcc upgrade.

**Comment by extrowerk** [comment]
I tried to make haiku.pkg with jam, as i can't seems to find any image generating rule wich sucess on x86_64. So : haiku.hpkg fails here with GCC8 on Haiku: http://chunk.io/miqlas/0bf5d713d94f4d44ab78ded2e906206a

**Comment by waddlesplash** [comment]
@extrowerk, that's because Haiku's tree is trying to use the GCC 5 syslibs package, not the GCC 8 one.

**Comment by waddlesplash** [comment]
Yes, but `haiku.hpkg` probably is buildable, the bootloader is in another package.

**Comment by waddlesplash** [comment]
Since we're using 7.3 and have made changes to that recipe, close as obsolete, and we'll revisit later on and adapt from the 7.3 recipe?

**Comment by extrowerk** [comment]
@waddlesplash : ok.

**Pull request closed without merging**